src/edu/csus/ecs/pc2/clics/API202306/EventFeedJSON.java

```diff
@@ -0,0 +1,1099 @@
+        fields[1] = account.getInstitutionName();
+        fields[2] = account.getInstitutionShortName();
+        return new CLICSOrganization(fields[0], account, fields).toJSON();
+    }
+
+    public String getTeamJSON(IInternalContest contest) {
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        Account[] accounts = EventFeedJSON.getTeamAccounts(contest);
+
+        if(isUseCollections()) {
+            StringBuilder dataCollection = new StringBuilder();
+            boolean bFirst = true;
+            dataCollection.append("[");
+            for (Account account : accounts) {
+                if (account.isAllowed(Permission.Type.DISPLAY_ON_SCOREBOARD) && isDisplayAccountGroupOnScoreboard(account) && (filter == null || filter.matches(account))) {
+                    if(bFirst) {
+                        bFirst = false;
+                    } else {
+                        dataCollection.append(",");
+                    }
+                    dataCollection.append(getTeamJSON(contest, account));
+                }
+            }
+            if(dataCollection.length() > 1) {
+                dataCollection.append("]");
+                appendJSONEvent(stringBuilder, TEAM_KEY, ++eventIdSequence, null, dataCollection.toString());
+                stringBuilder.append(NL);
+            }
+        } else {
+            for (Account account : accounts) {
+                if (account.isAllowed(Permission.Type.DISPLAY_ON_SCOREBOARD) && isDisplayAccountGroupOnScoreboard(account) && (filter == null || filter.matches(account))) {
+                    appendJSONEvent(stringBuilder, TEAM_KEY, ++eventIdSequence, IJSONTool.getAccountId(account), getTeamJSON(contest, account));
+                    stringBuilder.append(NL);
+                } else {
+                    ignoreTeam.add(account.getClientId());
+                }
+            }
+        }
+        return stringBuilder.toString();
+    }
+
+    public String getTeamJSON(IInternalContest contest, Account account) {
+        return new CLICSTeam(contest, account).toJSON();
+    }
+
+    public String getAccountJSON(IInternalContest contest) {
+
+        StringBuilder stringBuilder = new StringBuilder();
+
+        Account[] accounts = contest.getAccounts();
+        Arrays.sort(accounts, new AccountComparator());
+
+        if(isUseCollections()) {
+            StringBuilder dataCollection = new StringBuilder();
+            boolean bFirst = true;
+            dataCollection.append("[");
+            for (Account account : accounts) {
+                if(bFirst) {
+                    bFirst = false;
+                } else {
+                    dataCollection.append(",");
+                }
+                dataCollection.append(getTeamJSON(contest, account));
+            }
+            if(dataCollection.length() > 1) {
+                dataCollection.append("]");
```

Comment:
I don't understand why the closing `]` is ONLY added if the collection length is greater than 1.  It seems to me that since we're inside the `if(isUseCollections())` test, and we've definitely added an opening `[`, we would need to add a closing `]` _regardless of the length of the collection_.   For example, if there was only one account but we're still "using collections", the opening `[` gets added, then the (one) account gets added -- then don't we _always_ need to add the closing `]`?    

I notice that this same issue exists in other methods -- e.g. `getJudgementJSON()`, `getRunJSON()`, `getClarificationJSON()`, etc. -- so if I'm right that this is a bug then it needs to also be fixed in all those other methods...

Reply:
`dataCollection `accumulates the collection for the data objects being converted.  If anything was added to the collection (`length() > 1`), then it adds the `dataCollection` to the `stringBuilder `object.  If not, and `stringBuilder `remains an empty string which is returned and amounts to a no-op (no EF record is generated).